### PR TITLE
Polyhedron demo scale space plugin speedup sgiraudot

### PR DESCRIPTION
**Pull request by sgiraudot**
Only insert used points for polygon soups (speed-up and cleaner way of instanciating polygon soup).

Tested in https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.7-Ic-64.shtml

### CODE REVIEW

File: Installation/changes.html [hunk 1]
```diff
@@ -230,6 +230,14 @@ <h2 id="release4.7">Release 4.7 </h2>
...
   </ul>
 
+  
+  <h3>Polyhedron Demo (Scale Space Reconstruction Plugin)</h3>
+  <ul>
+    <li>Speed up construction of polygon soups by only inserting the
+      vertices that are actually used by each polygon soup.
+    </li>
+  </ul>
+
```
Comment (sloriot) [inline Installation/changes.html]: There is no need to update changes.html for a demo plugin update IMO

File: Polyhedron/demo/Polyhedron/Polyhedron_demo_scale_space_reconstruction_plugin.cpp [hunk 2]
```diff
@@ -103,18 +120,26 @@ void Polyhedron_demo_scale_space_reconstruction_plugin::on_actionScaleSpaceRecon
...
+        for (Reconstructor::Triple_iterator it=reconstruct.shell_begin( sh ),
                                            end=reconstruct.shell_end( sh );it!=end;++it)
         {
-          new_item->new_triangle( (*it)[0], (*it)[1], (*it)[2] );
+	  for (unsigned int ind = 0; ind < 3; ++ ind)
+          {
+	    if (map_i2i.find ((*it)[ind]) == map_i2i.end ())
+            {
+	      map_i2i.insert (std::make_pair ((*it)[ind], current_index ++));
+	      new_item->new_vertex (pts[(*it)[ind]].x (),
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/Polyhedron_demo_scale_space_reconstruction_plugin.cpp]: `find/insert` could be replaced by only one `insert`. Same remark below

**Comment by sloriot** [comment]
The should be merged again in integration, the windows testsuite was broken.

File: Installation/changes.html (hunk 1 above)
Comment (lrineau) [inline Installation/changes.html]: Do we ask Simon to remove the modification to changes.html, or it is acceptable that way?

Comment (sgiraudot) [inline Installation/changes.html]: Le 30/06/2015 11:24, Laurent Rineau a écrit :
>
> In Installation/changes.html 
> <https://github.com/CGAL/cgal/pull/144#discussion_r33554818>:
>
> > @@ -230,6 +230,14 @@ <h2 id="release4.7">Release 4.7 </h2>
> >      </li>
> >    </ul>
> >
> > +
> > +  <h3>Polyhedron Demo (Scale Space Reconstruction Plugin)</h3>
> > +  <ul>
> > +    <li>Speed up construction of polygon soups by only inserting the
> > +      vertices that are actually used by each polygon soup.
> > +    </li>
> > +  </ul>
> > +
> >
>
> Do we ask Simon to remove the modification to changes.html, or it is 
> acceptable that way?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/CGAL/cgal/pull/144/files#r33554818>.
>
I cancelled the modifications in changes.html this morning before 
pushing it into integration.

-- 
Simon Giraudot

**Comment by lrineau** [comment]
@sgiraudot Please push the updated branch to your Github fork, probably named `mine` if you have followed what is described in the wiki. That updates the pull-request.

See also ["Update a pull-request" on the wiki](https://cgal.geometryfactory.com/CGAL/Members/wiki/Guidelines/Source_Code_Management_GIT#Update_a_pull-request).

**Comment by lrineau** [comment]
I do not see the `scale_space_reconstruction_plugin` in the testsuite results, for example [here][here].

[here]: https://cgal.geometryfactory.com/CGAL/Members/testsuite/CGAL-4.7-Ic-72/Polyhedron_Demo/TestReport_lrineau_x86-64_Linux-Fedora19_g++-4.8_Release.gz

I think that the plugin is not listed in the commited script `cgal_test_with_cmake`.

For the moment, the [test results of `Polyhedron/demo` are fully red](https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.7-Ic-72.shtml#Polyhedron_Demo). I cannot merge this pull-request in `master`. I remerge it in `integration`...

**Comment by lrineau** [comment]
Have a look at the end of the file `Polyhedron/demo/Polyhedron/cgal_test_with_cmake`, in your branch. It should be straight-forward to add the plugin. You must add the name of the target: the name of the plugin, without the prefix `lib` and without the suffix `.so`/`.dll`.

**Comment by lrineau** [comment]
I have fixed the script `cgal_test_with_cmake` in the commit 2335fbc, pushed in the branch in `cgal-dev` and in your fork (and thus in this pull-request).

**Comment by sgiraudot** [comment]
Le 02/07/2015 15:46, Laurent Rineau a écrit :
>
> I have fixed the script |cgal_test_with_cmake| in the commit 2335fbc 
> <https://github.com/CGAL/cgal/commit/2335fbc7939cae7f2999ab85de5be3226c7726d2>, 
> pushed in the branch in |cgal-dev| and in your fork (and thus in this 
> pull-request).
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/CGAL/cgal/pull/144#issuecomment-118038304>.
>
Thanks!

-- 
Simon Giraudot

**Comment by lrineau** [comment]
I have also pushed a similar fix in the branch `sgiraudot/Scale_space_reconstruction_3-handle_non_manifold-sgiraudot,` (by error, I must admit).

**Comment by sloriot** [comment]
@sgiraudot What is the status of this branch?

**Comment by lrineau** [comment]
This pull request is waiting for a green line in the `Polyhedron_Demo` line of the testsuite.

**Pull request merged**